test/session.test.js

```diff
@@ -345,3 +345,114 @@ test('should update the expires property of the session using Session#touch() ev
 
   t.true(sessionExpires1 !== sessionExpires2)
 })
+
+test('should use custom sessionId generator if available (with request)', async (t) => {
+  const fastify = Fastify()
+  fastify.register(fastifyCookie)
+  fastify.register(fastifySession, {
+    secret: 'cNaoPYAwF60HZJzkcNaoPYAwF60HZJzk',
+    cookie: { secure: false, maxAge: 10000 },
+    idGenerator: (request) => {
+      if (request.session && request.session.returningVisitor) {
+        return `returningVisitor-${
+          new Date().getTime()
+        }-${
+          Math.random().toString().slice(2)
+        }`
+      }
+      return `custom-${
+        new Date().getTime()
+      }-${
+        Math.random().toString().slice(2)
+      }`
```

Comment:
Could you split this statement into multiple ones? (it is hard to read and there is some duplicated code